src/main/java/org/candlepin/controller/ProductManager.java

```diff
@@ -790,6 +787,32 @@ private static Product applyProductChanges(Product entity, ProductInfo update,
         return entity;
     }
 
+    /**
+     * Applies product attributes changes from the given ProductInfo instance to the specified
+     * Product entity. Attributes which have null values will be silently discarded from the
+     * updated attributes.
+     *
+     * @param entity
+     *  the Product entity to update
+     *
+     * @param update
+     *  the ProductInfo instance containing the data with which to update the entity
+     */
+    private static void applyProductAttributeChanges(Product entity, ProductInfo update) {
+        Map<String, String> incoming = update.getAttributes();
+        if (incoming != null) {
+            Map<String, String> filtered = new HashMap<>();
+
+            incoming.forEach((k, v) -> {
+                if (v != null) {
+                    filtered.put(k, v);
+                }
+            });
+
+            entity.setAttributes(filtered);
+        }
+    }
```

Comment:
Would it make more sense to put this in the setAttributes method on the Product class? That would help prevent someone from accidentally using the setAttributes without using the applyProductAttributeChanges and creating entries with null values.

Reply:
I spoke too soon. You already have the change in the Product setAttributes method!